expression/builtin_other.go

```diff
@@ -103,14 +134,59 @@ func (c *inFunctionClass) getFunction(ctx sessionctx.Context, args []Expression)
 	return sig, nil
 }
 
+type baseInSig struct {
+	baseBuiltinFunc
+	nonConstArgs []Expression
+	// hashSet will not be used when the number of constant arguments is smaller than threshold
+	threshold int
```

Comment:
Since the threshold is 1 or 2, can we always use the hash map when there is at least on constant in the argument list? The benefit is: code would be simpler, easier to be maintained